wasm/src/lib.rs

```diff
@@ -109,6 +123,46 @@ fn _get_qir(sources: SourceMap, language_features: LanguageFeatures) -> Result<S
     generate_qir(&store, package).map_err(|e| e.0.to_string())
 }
 
+fn _get_qir_preview(
+    sources: SourceMap,
+    language_features: LanguageFeatures,
+    profile: Profile,
+) -> Result<String, String> {
+    let core = compile::core();
+    let mut package_store = PackageStore::new(core);
+    let std = compile::std(&package_store, profile.into());
+    let std = package_store.insert(std);
+
+    let (unit, errors) = qsc::compile::compile(
+        &package_store,
+        &[std],
+        sources,
+        PackageType::Exe,
+        profile.into(),
+        language_features,
+    );
+
+    // Ensure it compiles before trying to add it to the store.
+    if !errors.is_empty() {
+        // This should never happen, as the program should be checked for errors before trying to
+        // generate code for it. But just in case, simply report the failure.
+        return Err("Failed to generate QIR".to_string());
+    }
+
+    let package_id = package_store.insert(unit);
+
+    let caps_results =
+        PassContext::run_fir_passes_on_hir(&package_store, package_id, profile.into());
+    // Ensure it compiles before trying to add it to the store.
+    if caps_results.is_err() {
+        // This should never happen, as the program should be checked for errors before trying to
+        // generate code for it. But just in case, simply report the failure.
+        return Err("Failed to generate QIR".to_string());
+    }
+
+    hir_to_qir(&package_store, package_id, profile.into()).map_err(|e| e.to_string())
```

Comment:
I think there might be an opportunity here to re-use the `PackageStoreComputeProperties` returned by RCA here. It seems like `run_fir_passes_on_hir` lowers to FIR, runs RCA and then the capababilities check pass. However, neither FIR or the results of RCA persist outside of this function so the `hir_to_qir` function has to lower to FIR and run RCA again.

I don't think this is a blocking issue, but it is worth considering opportunities where we don't have to repeat transformations and analysis if we don't have to.

Reply:
@cesarzc Good catch, updating.